src/map/utils/battleutils.cpp

```diff
@@ -4098,6 +4098,10 @@ namespace battleutils
             {
                 charmerBSTlevel = charmerBRDlevel;
             }
+            if (charmerBSTlevel > PCharmer->GetMLevel()) // takes level cap/sync into account
```

Comment:
`PCharmer->GetMLevel())` is already contained in the variable `charmerLvl`

this already gets factored into charm chance if you look at whats happening with the variable `base`.